src/eureka/S3_data_reduction/niriss.py

```diff
@@ -133,17 +133,30 @@ def get_wave(data, meta, log):
     log.writelog(f"  The NIRISS pupil position is {pwcpos:3f} degrees",
                  mute=(not meta.verbose))
 
-    norders = len(data.order)
+    norders = len(meta.all_orders)
     data['trace'] = (['x', 'order'],
                      np.zeros((data.x.shape[0], norders)) +
                      np.array(meta.src_ypos)[np.newaxis])
     data['wave_1d'] = (['x', 'order'],
                        np.zeros((data.x.shape[0], norders))*np.nan)
     data['wave_1d'].attrs['wave_units'] = 'microns'
 
-    for order in meta.orders:
+    for order in meta.all_orders:
         # Get trace for the given order and pupil position
         trace = get_soss_traces(pwcpos=pwcpos, order=str(order), interp=True)
+        if data.attrs['mhdr']['SUBARRAY'] == 'SUBSTRIP96' and \
+                meta.trace_offset is None:
+            # PASTASOSS doesn't account for different substrip starting rows;
+            # therefore, reset default trace offset to -10 pixels.
+            # SUBSTRIP96: SUBSTRT2 = 1803
+            # SUBSTRIP256: SUBSTRT2 = 1793
+            meta.trace_offset = -10
```

Comment:
Is it always precisely 10 pixels - your message seemed to suggest this wasn't always 10 px. Do we need something like the WFC3 2D-image correlation code?

Reply:
The value is constant for a given visit, but not between visits. So far, values of -11 or -12 have worked best for the visits I've looked at.